### PR TITLE
feat: enhance the `plugin status` command 

### PR DESCRIPTION
**Pull request by YanniHu1996**
closes: #3011

While working on this issue, I found that we have an opportunity to reduce the response time  of `status` API by the way used in this PR https://github.com/cloudnative-pg/cloudnative-pg/blob/af347f76cd234a1f715393c62fb6915904f7d159/pkg/management/postgres/probes.go#L66-L73 

Opened a new issue #3121 for tracing it

### CODE REVIEW

**Comment by mnencia** [comment]
@YanniHu1996 to fix the linter issues you should run `golangci-lint run --fix`

File: pkg/management/postgres/probes.go [hunk 1]
```diff
@@ -415,8 +510,8 @@ func (instance *Instance) fillWalStatusFromConnection(result *postgres.Postgresq
 			coalesce(sync_state, ''),
 			coalesce(sync_priority, 0)
 		FROM pg_catalog.pg_stat_replication
-		WHERE application_name LIKE $1 AND usename = $2`,
-		fmt.Sprintf("%s-%%", instance.ClusterName),
+		WHERE application_name ~ $1 AND usename = $2`,
+		fmt.Sprintf("%s-[0-9]+$", instance.ClusterName),
```
Comment (armru) [inline pkg/management/postgres/probes.go:514]: Personally I don't know if this changes should be in this patch

**Review by armru — APPROVED** [review]

File: pkg/management/postgres/probes.go [hunk 2]
```diff
@@ -270,9 +271,71 @@ func (instance *Instance) fillStatus(result *postgres.PostgresqlStatus) error {
...
+		FROM pg_stat_progress_basebackup b
+		   JOIN pg_stat_activity a USING (pid) 
+		WHERE application_name ~ '-join$'
+		ORDER BY 1, 2`)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if closeErr := rows.Close(); closeErr != nil && err == nil {
+			log.Error(err, "while closing rows")
```
Comment (litaocdl) [inline pkg/management/postgres/probes.go:311]: should we log the closeErr here

File: pkg/management/postgres/probes.go (hunk 1 above)
Comment (YanniHu1996) [reply pkg/management/postgres/probes.go:482]: the change to this SQL is to filter out rows that are not represented as replication to standby instances

**Review by leonardoce — APPROVED** [review]

**Comment by leonardoce** [comment]
We've problems with the PRs opened from repositories other than `cloudnative-pg/cloudnative-pg`. The build containers step is failing. See #3154

**Pull request merged**